test/e2e/cross_os/test_job_submissions.py

```diff
@@ -845,504 +1174,268 @@ def test_worker_always_runs_env_exit_despite_failure(
+                                ),
                             }
-                        ],
-                    }
-                )
-            )
-
-        config = configparser.ConfigParser()
+                        },
+                    },
+                ],
+            },
+        )
 
-        set_setting("defaults.farm_id", deadline_resources.farm.id, config)
-        set_setting("defaults.queue_id", deadline_resources.queue_a.id, config)
+        job.wait_until_complete(client=deadline_client)
+        logs_client: boto3.client = boto3.client(
+            "logs",
+            config=botocore.config.Config(retries={"max_attempts": 10, "mode": "adaptive"}),
+        )
 
-        job_id: Optional[str] = api.create_job_from_job_bundle(
-            job_bundle_path,
-            job_parameters,
-            priority=99,
-            max_retries_per_task=0,
-            config=config,
-            queue_parameter_definitions=[],
+        # Retrieve job output and verify the echo is printed
+        job_logs = job.get_logs(deadline_client=deadline_client, logs_client=logs_client)
+        full_log: str = "\n".join(
+            [le.message for _, log_events in job_logs.logs.items() for le in log_events]
         )
-        assert job_id is not None
 
-        job_details = Job.get_job_details(
-            client=deadline_client,
-            farm=deadline_resources.farm,
-            queue=deadline_resources.queue_a,
-            job_id=job_id,
+        assert (
+            full_log.count("HelloWorld") == 1
+        ), "Expected number of HelloWorld statements not found in job logs."
+
+        # Retrieve worker logs and verify that it's not empty
+        worker_log_group_name: str = (
+            f"/aws/deadline/{deadline_resources.farm.id}/{deadline_resources.fleet.id}"
         )
-        job = Job(
-            farm=deadline_resources.farm,
-            queue=deadline_resources.queue_a,
-            template={},
-            **job_details,
+        worker_id = session_worker.worker_id
+
+        worker_logs = logs_client.get_log_events(
+            logGroupName=worker_log_group_name,
+            logStreamName=worker_id,
+            startTime=int(job_start_time_seconds * 1000),
         )
 
-        # Query the session to check for progress percentage
-        complete_percentage: float = 0
+        assert len(worker_logs["events"]) > 0
 
-        @backoff.on_predicate(
-            wait_gen=backoff.constant,
-            max_time=60,
-            interval=2,
+    def test_worker_fails_session_action_timeout(
+        self,
+        deadline_resources: DeadlineResources,
+        deadline_client: DeadlineClient,
+    ) -> None:
+        # Test that if a task takes longer than the timeout defined, the session action goes to FAILED status
+        job = Job.submit(
```

Comment:
[`max_retries_per_task`](https://github.com/aws-deadline/deadline-cloud-test-fixtures/blob/adac342fc6dff6c6174ae39783448b1020cc9a61/src/deadline_test_fixtures/deadline/resources.py#L367C9-L367C29) should also probably be set to 1, rather than the default of 5.